deps/rabbitmq_auth_backend_oauth2/src/rabbit_auth_backend_oauth2.erl

```diff
@@ -186,29 +177,28 @@ check_token(DecodedToken) when is_map(DecodedToken) ->
     {ok, DecodedToken};
 
 check_token(Token) ->
-    Settings = application:get_all_env(?APP),
     case uaa_jwt:decode_and_verify(Token) of
-        {error, Reason} -> {refused, {error, Reason}};
-        {true, Payload} ->
-            validate_payload(post_process_payload(Payload, Settings));
-        {false, _}      -> {refused, signature_invalid}
+        {error, Reason} ->
+          {refused, {error, Reason}};
+        {true, TargetResourceServerId, Payload} ->
+          Payload0 = post_process_payload(TargetResourceServerId, Payload),
+          validate_payload(TargetResourceServerId, Payload0)
+%        _ -> {refused, signature_invalid}
```

Comment:
is the above pattern match exhaustive and we don't need a catch-all anymore?

Reply:
It is definitely needed !